lib/xfrm/sa.c

```diff
@@ -1700,6 +1713,22 @@ int xfrmnl_sa_set_family (struct xfrmnl_sa* sa, unsigned int family)
 	return 0;
 }
 
+int xfrmnl_sa_get_if_id (struct xfrmnl_sa* sa)
+{
+	if (sa->ce_mask & XFRM_SA_ATTR_IF_ID)
+		return sa->if_id;
+	else
+		return -1;
```

Comment:
return `-NLE_NOATTR`.

also, return-early and check for errors first:
```C
if (!(sa->ce_mask & XFRM_SA_ATTR_IF_ID)
     return -NLE_NOATTR;
return sa->if_id;
```

I understand that you copy&pasted this code, but we shouldn't follow bad patterns.

Reply:
Done. Also change to return value via passed in pointer to ensure we get full uint32_t

Reply:
oh right, this was important. Please change the `unsigned int` parameters in the public API to `uint32_t` types.

Reply:
Done. All public API use of if_id is now uint32_t